pkg/driver/nri_hooks.go

```diff
@@ -202,6 +202,14 @@ func (np *NetworkDriver) runPodSandbox(ctx context.Context, pod *api.PodSandbox)
 			klog.Infof("RunPodSandbox error configuring device %s namespace %s routing: %v", deviceName, ns, err)
 			return fmt.Errorf("error configuring device %s routes on namespace %s: %v", deviceName, ns, err)
 		}
+
+		// Configure neighbors
+		err = applyNeighborConfig(ns, ifNameInNs, config.NetworkInterfaceConfigInPod.Neighbors)
```

Comment:
Is there ever a time we may not want to have these neighbors programmed into the pods network? This as is will currently always take the permanent neighbors and add them into the pods entries.

Reply:
Yes you are correct that this will always add the permanent neighbors. Also, unless I misunderstood you, there's no way for the user to control that they "don't want permanent neighbor entries that we discovered". Any routes (and now neighbor entires as well) that the user provides in ResourceClaims configuration seems to be getting appended to the ones that we discover. Users don't get an override behaviour.

We certainly can't say that the "override" behaviour will never be needed. But I'd imagine the more common case being to copy the network interface as it was seen on the host's and the override behaviour to be a special case, which we can implement when needed.

Reply:
Yeah agreed for now, we just should aggregate it. If users need more granular control over the interfaces neighbors we can look into implementing it when we see a concrete use case.